skimage/measure/fit.py

```diff
@@ -492,66 +488,76 @@ def estimate(self, data):
         success : bool
             True, if model estimation succeeds.
 
-        """
 
+        References:
+        -----------
+            [1] Halir, R., Flusser, J.: 'Numerically Stable Direct Least Squares
+                Fitting of Ellipses'
+            [2] http://mathworld.wolfram.com/Ellipse.html
+
+        """
         _check_data_dim(data, dim=2)
 
         x = data[:, 0]
         y = data[:, 1]
 
-        N = data.shape[0]
-
-        # pre-allocate jacobian for all iterations
-        A = np.zeros((N + 5, 2 * N), dtype=np.double)
-        # same for all iterations: xc, yc
-        A[0, :N] = -1
-        A[1, N:] = -1
-
-        diag_idxs = np.diag_indices(N)
-
-        def fun(params):
-            xyt = self.predict_xy(params[5:], params[:5])
-            fx = x - xyt[:, 0]
-            fy = y - xyt[:, 1]
-            return np.append(fx, fy)
-
-        def Dfun(params):
-            xc, yc, a, b, theta = params[:5]
-            t = params[5:]
-
-            ct = np.cos(t)
-            st = np.sin(t)
-            ctheta = math.cos(theta)
-            stheta = math.sin(theta)
-
-            # derivatives for fx, fy in the following order:
-            #       xc, yc, a, b, theta, t_i
-
-            # fx
-            A[2, :N] = - ctheta * ct
-            A[3, :N] = stheta * st
-            A[4, :N] = a * stheta * ct + b * ctheta * st
-            A[5:, :N][diag_idxs] = a * ctheta * st + b * stheta * ct
-            # fy
-            A[2, N:] = - stheta * ct
-            A[3, N:] = - ctheta * st
-            A[4, N:] = - a * ctheta * ct + b * stheta * st
-            A[5:, N:][diag_idxs] = a * stheta * st - b * ctheta * ct
-
-            return A
-
-        # initial guess of parameters using a circle model
-        params0 = np.empty((N + 5, ), dtype=np.double)
-        xc0 = x.mean()
-        yc0 = y.mean()
-        r0 = np.sqrt((x - xc0)**2 + (y - yc0)**2).mean()
-        params0[:5] = (xc0, yc0, r0, 0, 0)
-        params0[5:] = np.arctan2(y - yc0, x - xc0)
-
-        params, _ = optimize.leastsq(fun, params0, Dfun=Dfun, col_deriv=True)
-
-        self.params = params[:5]
-
+        # Quadratic part of design matrix [eqn. 15] from [1]
+        D1 = np.mat(np.vstack([x ** 2, x * y, y ** 2])).T
```

Comment:
We discourage the use of `numpy.matrix` objects at scikit-image. (I might be mistaken on this but if that's the case, someone else should jump in and correct me.) The reason is that these objects are very similar but critically different to the far more common numpy arrays. Therefore, a reader seeing a matrix object being used might incorrectly think it's an array, and incorrectly interpret the use of e.g. the `*` operator.

I know it's uglier, but please change the code to use numpy arrays everywhere and `np.dot` for matrix multiplication. (At some indeterminate point in the future, we can move to Python 3.5+ and use the `@` operator to get both unambiguity and elegance.)

Some other devs might want to jump in and opine here... @emmanuelle @stefanv @ahojnnes ?